### PR TITLE
fix: guard api_plugin_moveup/movedown against NULL prior/next id (1.2.x backport)

### PR DESCRIPTION
**Pull request by somethingwithproof**
Backport of #7157 to 1.2.x. The same two bugs exist identically in both branches.

## Root cause

`api_plugin_moveup()` / `api_plugin_movedown()` rotate plugin ids via a three-step swap. At the boundary (first plugin moved up, last moved down), `MAX(id) WHERE id < ?` / `MIN(id) WHERE id > ?` returns `NULL`. With strict SQL modes stripped on connect, `UPDATE plugin_config SET id = NULL` on a `NOT NULL` column silently stores `0`, corrupting the primary key. Subsequent calls to `plugins_load_temp_table()` then fail with `ERROR 1062` because the id=0 row gets AUTO_INCREMENT-reassigned to the next sequence value, colliding with the row already at that id.

## Fixes

- `api_plugin_moveup`: add `!empty($prior_id)` guard — moving the first plugin is a no-op.
- `api_plugin_movedown`: add outer `!empty($id)` guard (was missing) and `!empty($next_id)` guard — moving the last plugin is a no-op.
- `plugins_load_temp_table`: set `NO_AUTO_VALUE_ON_ZERO` for the single bulk copy INSERT, then restore the original session sql_mode, so any existing id=0 row is preserved literally.

Includes `tests/Unit/PluginMoveorderBugTest.php` with 12 regression tests (all pass).

### CODE REVIEW

**Review by TheWitness — DISMISSED** [review]

File: lib/plugins.php [hunk 1]
```diff
@@ -988,31 +988,41 @@ function api_plugin_moveup($plugin) {
...
-		db_execute_prepared('UPDATE plugin_config SET id = ? WHERE id = ?', array($prior_id, $id));
-		db_execute_prepared('UPDATE plugin_config SET id = ? WHERE id = ?', array($id, $temp_id));
+		/* MAX() on an empty set returns NULL; without a guard, the UPDATE
+		 * below would set id = NULL on the current plugin, which non-strict
+		 * MariaDB/MySQL silently stores as 0, corrupting the primary key. */
+		if (!empty($prior_id)) {
+			$temp_id = db_fetch_cell('SELECT MAX(id) FROM plugin_config') + 1;
+
+			db_execute_prepared('UPDATE plugin_config SET id = ? WHERE id = ?', array($temp_id, $prior_id));
+			db_execute_prepared('UPDATE plugin_config SET id = ? WHERE id = ?', array($prior_id, $id));
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/plugins.php]: Fixed — replaced all empty() guards with strict !== null (MAX/MIN results) and !== false (db_fetch_cell_prepared no-row results). Test assertions updated to match.

File: tests/Unit/PluginMoveorderBugTest.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+test('api_plugin_moveup has !empty($prior_id) guard around the three-step swap', function () use ($libPluginsPath) {
+	$source = file_get_contents($libPluginsPath);
+
+	$fn_pos    = strpos($source, 'function api_plugin_moveup(');
+	$guard_pos = strpos($source, 'if (!empty($prior_id))', $fn_pos);
+
+	expect($fn_pos)->not->toBeFalse('api_plugin_moveup not found');
+	expect($guard_pos)->not->toBeFalse('!empty($prior_id) guard missing from api_plugin_moveup');
+});
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/Unit/PluginMoveorderBugTest.php]: Fixed — test assertions updated to search for the new strict-comparison guard strings.

File: plugins.php [hunk 3]
```diff
@@ -235,7 +235,15 @@ function plugins_load_temp_table() {
 
 			db_execute("CREATE TEMPORARY TABLE IF NOT EXISTS $table LIKE plugin_config");
 			db_execute("TRUNCATE $table");
+
+			/* Cacti strips NO_AUTO_VALUE_ON_ZERO on connect (database.php). Without it,
+			 * a row with id=0 in plugin_config (e.g. from a plugin upgrade script) is
+			 * reassigned by AUTO_INCREMENT to the next sequence value, causing a 1062
+			 * collision when another row already holds that id. */
+			$orig_sql_mode = db_fetch_cell('SELECT @@SESSION.sql_mode');
+			db_execute("SET SESSION sql_mode = CONCAT_WS(',', @@SESSION.sql_mode, 'NO_AUTO_VALUE_ON_ZERO')");
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply plugins.php]: Fixed — sql_mode is now built in PHP: split $orig_sql_mode on commas, trim and filter empty tokens, append NO_AUTO_VALUE_ON_ZERO only if absent, then set via db_execute_prepared. Handles both empty-string and leading/trailing-comma edge cases.

File: lib/plugins.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/plugins.php]: Fixed.

File: tests/Unit/PluginMoveorderBugTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/Unit/PluginMoveorderBugTest.php]: Fixed.

File: plugins.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply plugins.php]: Fixed.

**Review by TheWitness — APPROVED** [review]
